keps/sig-windows/3503-host-network-support-for-windows-pods/README.md

```diff
@@ -0,0 +1,861 @@
+
+- E2E tests will be added to validate this functionality.
+
+<!--
+ - <test>: <link to test coverage>
+-->
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, [feature gate] graduations, or as
+something else. The KEP should keep this high-level with a focus on what
+signals will be looked at to determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- CRI-API updates added to codebase
+- Unit tests added to validate correct CRI-API fields are set depending if `hostNetwork=true` is set on Windows
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- A version of containerd w/ support for configuing pod's to use the node's network namespace is released (target v1.8)
+
+#### GA
+
+- All feedback from alpha/beta usage is addressed
```

Comment:
should this wait until all the runtimes kubernetes supported versions support the feature?

otherwise, you can have a feature enabled that will not work, because the runtime implementation doesn't support it

Reply:
CRI-O doesn't currently support Windows so that just leaves containerd :)

Reply:
this makes things simpler

Reply:
but you still have to wait for containerd to EOL versions without the support, no?

Reply:
For other SIG-Windows enhancements that required containerd updates we have not waited for versions of containerd w/o support to EOL.
We instead relied on documentation and heavy messaging and that has been sufficient. 

I believe containerd only supports 2 minor versions and releases about every 6-9 months so if needed we could add this as a graduation criterion if needed (it just hasn't been an issue in the past).

Reply:
it is really your call, thanks for clarifying